manifests/params.pp

```diff
@@ -58,4 +60,13 @@
       fail("Module ${module_name} is not supported on osfamily '${::osfamily}'")
     }
   }
+	case $::architecture {
+    'x86_64', 'amd64': { $arch = 'amd64' }
+    'i386':            { $arch = '386'   }
+    /^arm.*/:          { $arch = 'arm'   }
+    default:           {
+      fail("Unsupported kernel architecture: ${::architecture}")
```

Comment:
@jsok under further review, the module currently only has support for Debian and Redhat. I know Debian is fairly renowned for supporting lots of different arch/kernels, but sane defaults makes more sense? I am happy to check for the presence of the `download_url` inside the case statement, but I expect the code to get very, very unreadable after a while.